src/Features/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -323,12 +383,13 @@ private async Task<bool> DoesAllTextMatchWorkspaceSolutionAsync(ImmutableDiction
         return true;
     }
 
-    private static async Task<bool> AreChecksumsEqualAsync(Document document, SourceText lspText, CancellationToken cancellationToken)
+    private static async ValueTask<bool> AreChecksumsEqualAsync(Document document, SourceText lspText, CancellationToken cancellationToken)
     {
-        var documentText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
-        var documentTextChecksum = documentText.GetChecksum();
-        var lspTextChecksum = lspText.GetChecksum();
-        return lspTextChecksum.SequenceEqual(documentTextChecksum);
+        var documentText = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
+        if (documentText == lspText)
+            return true;
+
+        return lspText.GetChecksum().AsSpan().SequenceEqual(documentText.GetChecksum().AsSpan());
```

Comment:
AsSpan().SequenceEqual calls into a specialized overload that optimizes for the Span<byte> case.